python/ec2_vpc_public_subnet.py

```diff
@@ -0,0 +1,95 @@
+#
+# This file made available under CC0 1.0 Universal (https://creativecommons.org/publicdomain/zero/1.0/legalcode)
+# 
+# Description: Check that no EC2 are in Public Subnet
+#
+# Trigger Type: Change Triggered
+# Scope of Changes: EC2:Instance
+# Accepted Parameters: None
+# Your Lambda function execution role will need to have a policy that provides the appropriate
+# permissions.  Here is a policy that you can consider.  You should validate this for your own
+# environment
+#{
+#    "Version": "2012-10-17",
+#    "Statement": [
+#        {
+#            "Effect": "Allow",
+#            "Action": [
+#                "logs:CreateLogGroup",
+#                "logs:CreateLogStream",
+#                "logs:PutLogEvents"
+#            ],
+#            "Resource": "arn:aws:logs:*:*:*"
+#        },
+#        {
+#            "Effect": "Allow",
+#            "Action": [
+#                "config:PutEvaluations",
+#                "ec2:DescribeRouteTables"
+#            ],
+#            "Resource": "*"
+#        }
+#    ]
+#}
+#
+
+import boto3
+import botocore
+import json
+import logging
+
+log = logging.getLogger()
+log.setLevel(logging.INFO)
+
+def evaluate_compliance(configuration_item):
+    subnet_id   = configuration_item["configuration"]["subnetId"]
+    vpc_id      = configuration_item["configuration"]["vpcId"]
+    client      = boto3.client("ec2");
+    private     = True
+
+    response    = client.describe_route_tables()
```

Comment:
I've added describe_route_tables to filter with route.destination-cidr-block having 0.0.0.0/0
 https://github.com/PriceBoardIn/aws-config-rules/commit/b67f0a687adc0d4465144162a722e5eb238f4833